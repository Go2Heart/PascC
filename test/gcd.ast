©¸©¤©¤ program    ©¸©¤©¤ ('program_head', 'example')        ©¸©¤©¤ idlist            ©¸©¤©¤ ('id', 'input')            ©¸©¤©¤ ('id', 'output')    ©¸©¤©¤ program_body        ©¸©¤©¤ const_declarations            ©¸©¤©¤ ('const_declaration',)                ©¸©¤©¤ ('id', 'test_char')                ©¸©¤©¤ ('char', "'c'")            ©¸©¤©¤ ('const_declaration',)                ©¸©¤©¤ ('id', 'test_integer')                ©¸©¤©¤ ('integer', 1)            ©¸©¤©¤ ('const_declaration',)                ©¸©¤©¤ ('id', 'test_minus_integer')                ©¸©¤©¤ ('integer', '-1')        ©¸©¤©¤ var_declarations            ©¸©¤©¤ var_declaration                ©¸©¤©¤ ('id', 'x')                ©¸©¤©¤ ('id', 'y')                ©¸©¤©¤ ('integer',)            ©¸©¤©¤ var_declaration                ©¸©¤©¤ ('id', 'c')                ©¸©¤©¤ ('array_type',)                    ©¸©¤©¤ period                        ©¸©¤©¤ ('const_value', 1)                        ©¸©¤©¤ ('const_value', 2)                        ©¸©¤©¤ ('const_value', 5)                        ©¸©¤©¤ ('const_value', 6)                    ©¸©¤©¤ ('char',)        ©¸©¤©¤ subprogram_declarations            ©¸©¤©¤ subprogram                ©¸©¤©¤ ('function_head', 'gcd')                    ©¸©¤©¤ formal_parameter                        ©¸©¤©¤ parameter_list                            ©¸©¤©¤ value_parameter                                ©¸©¤©¤ ('id', 'a')                                ©¸©¤©¤ ('id', 'b')                                ©¸©¤©¤ ('integer',)                            ©¸©¤©¤ var_parameter                                ©¸©¤©¤ value_parameter                                    ©¸©¤©¤ ('id', 'c')                                    ©¸©¤©¤ ('id', 'd')                                    ©¸©¤©¤ ('integer',)                    ©¸©¤©¤ ('integer',)                ©¸©¤©¤ subprogram_body                    ©¸©¤©¤ compound_statement                        ©¸©¤©¤ statement_list                            ©¸©¤©¤ compound_statement                                ©¸©¤©¤ statement_list                                    ©¸©¤©¤ assignment_statement                                        ©¸©¤©¤ fun_id                                        ©¸©¤©¤ expression                                            ©¸©¤©¤ simple_expression                                                ©¸©¤©¤ term                                                    ©¸©¤©¤ factor                                                        ©¸©¤©¤ ('number', 0)                                    ©¸©¤©¤ empty_statement                            ©¸©¤©¤ if_statement                                ©¸©¤©¤ expression                                    ©¸©¤©¤ simple_expression                                        ©¸©¤©¤ term                                            ©¸©¤©¤ factor                                                ©¸©¤©¤ variable                                                    ©¸©¤©¤ ('id', 'b')                                    ©¸©¤©¤ simple_expression                                        ©¸©¤©¤ term                                            ©¸©¤©¤ factor                                                ©¸©¤©¤ ('number', 0)                                ©¸©¤©¤ assignment_statement                                    ©¸©¤©¤ fun_id                                    ©¸©¤©¤ expression                                        ©¸©¤©¤ simple_expression                                            ©¸©¤©¤ term                                                ©¸©¤©¤ factor                                                    ©¸©¤©¤ variable                                                        ©¸©¤©¤ ('id', 'a')                                ©¸©¤©¤ assignment_statement                                    ©¸©¤©¤ fun_id                                    ©¸©¤©¤ expression                                        ©¸©¤©¤ simple_expression                                            ©¸©¤©¤ term                                                ©¸©¤©¤ factor                                                    ©¸©¤©¤ ('id', 'gcd')                                                    ©¸©¤©¤ expression_list                                                        ©¸©¤©¤ expression                                                            ©¸©¤©¤ simple_expression                                                                ©¸©¤©¤ term                                                                    ©¸©¤©¤ factor                                                                        ©¸©¤©¤ variable                                                                            ©¸©¤©¤ ('id', 'b')                                                        ©¸©¤©¤ expression                                                            ©¸©¤©¤ simple_expression                                                                ©¸©¤©¤ term                                                                    ©¸©¤©¤ term                                                                        ©¸©¤©¤ factor                                                                            ©¸©¤©¤ variable                                                                                ©¸©¤©¤ ('id', 'a')                                                                    ©¸©¤©¤ factor                                                                        ©¸©¤©¤ variable                                                                            ©¸©¤©¤ ('id', 'b')            ©¸©¤©¤ subprogram                ©¸©¤©¤ ('procedure_head', 'gcd1')                    ©¸©¤©¤ formal_parameter                        ©¸©¤©¤ parameter_list                            ©¸©¤©¤ value_parameter                                ©¸©¤©¤ ('id', 'a')                                ©¸©¤©¤ ('id', 'b')                                ©¸©¤©¤ ('integer',)                ©¸©¤©¤ subprogram_body                    ©¸©¤©¤ compound_statement                        ©¸©¤©¤ statement_list                            ©¸©¤©¤ if_statement                                ©¸©¤©¤ expression                                    ©¸©¤©¤ simple_expression                                        ©¸©¤©¤ term                                            ©¸©¤©¤ factor                                                ©¸©¤©¤ variable                                                    ©¸©¤©¤ ('id', 'b')                                    ©¸©¤©¤ simple_expression                                        ©¸©¤©¤ term                                            ©¸©¤©¤ factor                                                ©¸©¤©¤ ('number', 0)                                ©¸©¤©¤ assignment_statement                                    ©¸©¤©¤ fun_id                                    ©¸©¤©¤ expression                                        ©¸©¤©¤ simple_expression                                            ©¸©¤©¤ term                                                ©¸©¤©¤ factor                                                    ©¸©¤©¤ variable                                                        ©¸©¤©¤ ('id', 'a')                                ©¸©¤©¤ assignment_statement                                    ©¸©¤©¤ fun_id                                    ©¸©¤©¤ expression                                        ©¸©¤©¤ simple_expression                                            ©¸©¤©¤ term                                                ©¸©¤©¤ factor                                                    ©¸©¤©¤ ('id', 'gcd')                                                    ©¸©¤©¤ expression_list                                                        ©¸©¤©¤ expression                                                            ©¸©¤©¤ simple_expression                                                                ©¸©¤©¤ term                                                                    ©¸©¤©¤ factor                                                                        ©¸©¤©¤ variable                                                                            ©¸©¤©¤ ('id', 'b')                                                        ©¸©¤©¤ expression                                                            ©¸©¤©¤ simple_expression                                                                ©¸©¤©¤ term                                                                    ©¸©¤©¤ term                                                                        ©¸©¤©¤ factor                                                                            ©¸©¤©¤ variable                                                                                ©¸©¤©¤ ('id', 'a')                                                                    ©¸©¤©¤ factor                                                                        ©¸©¤©¤ variable                                                                            ©¸©¤©¤ ('id', 'b')        ©¸©¤©¤ compound_statement            ©¸©¤©¤ statement_list                ©¸©¤©¤ read_statement                    ©¸©¤©¤ variable_list                        ©¸©¤©¤ variable                            ©¸©¤©¤ ('id', 'x')                        ©¸©¤©¤ variable                            ©¸©¤©¤ ('id', 'y')                ©¸©¤©¤ write_statement                    ©¸©¤©¤ expression_list                        ©¸©¤©¤ expression                            ©¸©¤©¤ simple_expression                                ©¸©¤©¤ term                                    ©¸©¤©¤ factor                                        ©¸©¤©¤ ('id', 'gcd')                                        ©¸©¤©¤ expression_list                                            ©¸©¤©¤ expression                                                ©¸©¤©¤ simple_expression                                                    ©¸©¤©¤ term                                                        ©¸©¤©¤ factor                                                            ©¸©¤©¤ variable                                                                ©¸©¤©¤ ('id', 'x')                                            ©¸©¤©¤ expression                                                ©¸©¤©¤ simple_expression                                                    ©¸©¤©¤ term                                                        ©¸©¤©¤ factor                                                            ©¸©¤©¤ variable                                                                ©¸©¤©¤ ('id', 'y')