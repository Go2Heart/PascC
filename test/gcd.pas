
program example(input,output);
    const z=500;p=250;

    var x,y:integer;
        c:array[1..2,5..6] of char;
    function gcd(a,b:integer;var c,d:integer):integer;
        begin 
            if b=0 then gcd:=a
            else gcd:=gcd(b, a mod b)
        end;
    procedure gcd1(a,b:integer);
        begin 
            if b=0 then gcd:=a
            else gcd:=gcd(b, a mod b)
        end;
    begin
        read(x, y);
        write(gcd(x, y))
    end.