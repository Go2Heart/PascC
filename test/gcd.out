LexToken(PROGRAM,'program',2,1)
LexToken(ID,'example',2,9)
LexToken(LPAREN,'(',2,16)
LexToken(ID,'input',2,17)
LexToken(COMMA,',',2,22)
LexToken(ID,'output',2,23)
LexToken(RPAREN,')',2,29)
LexToken(SEMI,';',2,30)
LexToken(VAR,'var',3,36)
LexToken(ID,'x',3,40)
LexToken(COMMA,',',3,41)
LexToken(ID,'y',3,42)
LexToken(COLON,':',3,43)
LexToken(INTEGER,'integer',3,44)
LexToken(SEMI,';',3,51)
LexToken(FUNCTION,'function',4,57)
LexToken(ID,'gcd',4,66)
LexToken(LPAREN,'(',4,69)
LexToken(ID,'a',4,70)
LexToken(COMMA,',',4,71)
LexToken(ID,'b',4,72)
LexToken(COLON,':',4,73)
LexToken(INTEGER,'integer',4,74)
LexToken(RPAREN,')',4,81)
LexToken(COLON,':',4,82)
LexToken(INTEGER,'integer',4,83)
LexToken(SEMI,';',4,90)
LexToken(BEGIN,'begin',5,100)
LexToken(IF,'if',6,119)
LexToken(ID,'b',6,122)
LexToken(EQU,'=',6,123)
LexToken(ICONST,0,6,124)
LexToken(THEN,'then',6,126)
LexToken(ID,'gcd',6,131)
LexToken(ASSIGN,':=',6,134)
LexToken(ID,'a',6,136)
LexToken(ELSE,'else',7,150)
LexToken(ID,'gcd',7,155)
LexToken(ASSIGN,':=',7,158)
LexToken(ID,'gcd',7,160)
LexToken(LPAREN,'(',7,163)
LexToken(ID,'b',7,164)
LexToken(COMMA,',',7,165)
LexToken(ID,'a',7,167)
LexToken(MULDIVANDOP,'mod',7,169)
LexToken(ID,'b',7,173)
LexToken(RPAREN,')',7,174)
LexToken(END,'end',8,184)
LexToken(SEMI,';',8,187)
LexToken(BEGIN,'begin',9,193)
LexToken(READ,'read',10,207)
LexToken(LPAREN,'(',10,211)
LexToken(ID,'x',10,212)
LexToken(COMMA,',',10,213)
LexToken(ID,'y',10,215)
LexToken(RPAREN,')',10,216)
LexToken(SEMI,';',10,217)
LexToken(WRITE,'write',11,227)
LexToken(LPAREN,'(',11,232)
LexToken(ID,'gcd',11,233)
LexToken(LPAREN,'(',11,236)
LexToken(ID,'x',11,237)
LexToken(COMMA,',',11,238)
LexToken(ID,'y',11,240)
LexToken(RPAREN,')',11,241)
LexToken(RPAREN,')',11,242)
LexToken(END,'end',12,248)
LexToken(DOT,'.',12,251)
